### PR TITLE
Fix isNaN check

### PR DESCRIPTION
**Pull request by joshbetz**
parseInt will return an int if the first character is a number. If we pass in a name that starts with a number, we don't want it to be interpreted as a number.

### CODE REVIEW

File: src/lib/api/app.js [hunk 1]
```diff
@@ -16,7 +16,8 @@ export default async function( app: string | number, fields: ?any ): Promise<any
 	}
 
 	const api = await API();
-	if ( isNaN( parseInt( app ) ) ) {
+	if ( isNaN( app ) ) {
+		app = parseInt( app, 10 );
```
Comment (mjangda) [inline src/lib/api/app.js]: I don't think we want the `parseInt` here since it's the app `name` (a string). We need it for the `id` condition below.

**Review by mjangda — CHANGES_REQUESTED** [review]

File: src/lib/api/app.js (hunk 1 above)
Comment (joshbetz) [reply src/lib/api/app.js]: Oh, good catch. I guess I only tested with an ID 🙃

**Comment by joshbetz** [comment]
I guess this test won't work because it requires a valid token 🤔

**Comment by mjangda** [comment]
We can mock `api.query` and/or use nock to mock API requests.

**Comment by joshbetz** [comment]
Looks like we can mock GraphQL queries with graphql-tools. https://www.apollographql.com/docs/graphql-tools/mocking.html

I think the problem with using nock here would be that every request is to `/graphql`. We could mock requests to `/graphql`, but this test is not that interesting if we hardcode the API response.

**Pull request merged**